keyboards/2d4d/rules.mk

```diff
@@ -0,0 +1,22 @@
+# MCU name
+MCU = STM32F401
+BOARD = BLACKPILL_STM32_F401
+
+# Bootloader selection
+BOOTLOADER = stm32-dfu
+
+# Build Options
+#   change yes to no to disable
+#
+BOOTMAGIC_ENABLE = yes # Enable Bootmagic Lite
+MOUSEKEY_ENABLE = yes  # Mouse keys
+EXTRAKEY_ENABLE = yes  # Audio control and System control
+CONSOLE_ENABLE = no    # Console for debug
+COMMAND_ENABLE = no    # Commands for debug and configuration
+NKRO_ENABLE = no      # Enable N-Key Rollover
+BACKLIGHT_ENABLE = no  # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = no  # Enable keyboard RGB underglow
+AUDIO_ENABLE = no      # Audio output
+ENCODER_ENABLE = yes   # Enable encoder
```

Comment:
If encoders are enabled at the keyboard level, there should exist some functionality for them at the keyboard level as well, that way it works for QMK Configurator users. See [doc](https://docs.qmk.fm/#/feature_encoders?id=callbacks).